### ISSUE TITLE
KeyVault set_secret_async referencing SecretSetParameters incorrectly

### ISSUE THREAD

**Issue by jstange** [Azure/azure-sdk-for-ruby#2831]
The 7.1 client (and probably the other api versions) as of `azure_key_vault-0.18.1` try to instantiate `SecretSetParameters` without a scope, which quietly returns a `nil`, and also gets the name of one of the parameters wrong if you do get a real object out. The following diff fixes it for me:

```
diff --git a/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb b/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb
index d6fe873..3a66e9f 100644
--- a/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb
+++ b/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb
@@ -6000,12 +6000,12 @@ module Azure::KeyVault::V7_1
       fail ArgumentError, 'api_version is nil' if api_version.nil?
       fail ArgumentError, 'value is nil' if value.nil?

-      parameters = SecretSetParameters.new
+      parameters = Azure::KeyVault::V7_1::Models::SecretSetParameters.new
       unless value.nil? && tags.nil? && content_type.nil? && secret_attributes.nil?
         parameters.value = value
         parameters.tags = tags
         parameters.content_type = content_type
-        parameters.secretAttributes = secret_attributes
+        parameters.secret_attributes = secret_attributes
       end

       request_headers = {}

```

**Comment by kairu-ms** [comment]
Thanks for your feedback. I've manually fixed above issues. Please try 0.19.0. BTW this repo is [retired](https://github.com/Azure/azure-sdk-for-ruby#important-announcement).

**Comment by serprex** [comment]
Not fixed in 0.19.0 is `set_certificate_issuer_async` which still suffers from the same class of issue described here, currently working around it with our own fork which has
```diff
diff --git a/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb b/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb
index 8921998e9..67493b34a 100644
--- a/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb
+++ b/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb
@@ -843,11 +843,11 @@ module Azure::KeyVault::V7_1
       fail ArgumentError, 'api_version is nil' if api_version.nil?
       fail ArgumentError, 'provider is nil' if provider.nil?

-      parameter = CertificateIssuerSetParameters.new
+      parameter = Models::CertificateIssuerSetParameters.new
       unless provider.nil? && credentials.nil? && organization_details.nil? && attributes.nil?
         parameter.provider = provider
         parameter.credentials = credentials
-        parameter.OrganizationDetails = organization_details
+        parameter.organization_details = organization_details
         parameter.attributes = attributes
       end
```

**Comment by kairu-ms** [comment]
> Not fixed in 0.19.0 is `set_certificate_issuer_async` which still suffers from the same class of issue described here, currently working around it with our own fork which has
> 
> ```diff
> diff --git a/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb b/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb
> index 8921998e9..67493b34a 100644
> --- a/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb
> +++ b/data/azure_key_vault/lib/7.1/generated/azure_key_vault/key_vault_client.rb
> @@ -843,11 +843,11 @@ module Azure::KeyVault::V7_1
>        fail ArgumentError, 'api_version is nil' if api_version.nil?
>        fail ArgumentError, 'provider is nil' if provider.nil?
> 
> -      parameter = CertificateIssuerSetParameters.new
> +      parameter = Models::CertificateIssuerSetParameters.new
>        unless provider.nil? && credentials.nil? && organization_details.nil? && attributes.nil?
>          parameter.provider = provider
>          parameter.credentials = credentials
> -        parameter.OrganizationDetails = organization_details
> +        parameter.organization_details = organization_details
>          parameter.attributes = attributes
>        end
> ```

Thanks for your feedback. I will release a new version to fix it soon.
